typings/index.d.ts

```diff
@@ -378,6 +379,239 @@ declare module 'discord.js' {
 		public once(event: 'end', listener: (collected: Collection<K, V>, reason: string) => void): this;
 	}
 
+	type AllowedImageFormat = 'webp' | 'png' | 'jpg' | 'gif';
+
+	export interface Constants {
+		Package: {
+			name: string;
+			version: string;
+			description: string;
+			author: string;
+			license: string;
+			main: PathLike;
+			types: PathLike;
+			homepage: string;
+			keywords: string[];
+			bugs: { url: string };
+			repository: { type: string, url: string };
+			browser: { [key: string]: boolean };
+			scripts: { [key: string]: string };
+			engines: { [key: string]: string };
+			dependencies: { [key: string]: string };
+			peerDependencies: { [key: string]: string };
+			devDependencies: { [key: string]: string };
+			[key: string]: any;
+		};
+		browser: boolean;
+		DefaultOptions: ClientOptions;
+		UserAgent: string | null;
+		Endpoints: {
+			botGateway: string;
+			invite: (root: string, code: string) => string;
+			CDN: (root: string) => {
+				Asset: (name: string) => string;
+				DefaultAvatar: (id: string | number) => string;
+				Emoji: (emojiID: string, format: 'png' | 'gif') => string;
+				Avatar: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				Icon: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				AppIcon: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				AppAsset: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				GDMIcon: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+				Splash: (userID: string | number, hash: string, format: AllowedImageFormat, size: number) => string;
+			};
+		};
+		WSCodes: {
+			1000: 'Connection gracefully closed';
+			4004: 'Tried to identify with an invalid token';
+			4010: 'Sharding data provided was invalid';
+			4011: 'Shard would be on too many guilds if connected';
+		};
+		Events: {
+			RATE_LIMIT: 'rateLimit';
+			READY: 'ready';
+			RESUMED: 'resumed';
+			GUILD_CREATE: 'guildCreate';
+			GUILD_DELETE: 'guildDelete';
+			GUILD_UPDATE: 'guildUpdate';
+			GUILD_UNAVAILABLE: 'guildUnavailable';
+			GUILD_AVAILABLE: 'guildAvailable';
```

Comment:
While being in the constants this event is not actually emitted from within d.js.